examples/systemcalls/systemcalls.c

```diff
@@ -83,17 +106,43 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
     // this line is to avoid a compile warning before your implementation is complete
     // and may be removed
     command[count] = command[count];
+    int status;
+    int fd = open(outputfile, O_RDWR|O_CREAT, 0644);
+    if (fd < 0) 
+    { perror("open"); 
+    exit(-1);
+    }
 
+    pid=fork();
+    if(pid==-1){
+        perror("fork");
+        return false;
+    }
+    else if(pid==0){ 
+   if (dup2(fd, 1) < 0){ 
+    perror("dup2"); 
+    exit(-1); }
+    close(fd);
+    ret=execv(command[0],command);
+    if(ret == -1){
+    perror("execv");
+    exit (EXIT_FAILURE);
+    }
+    exit (EXIT_FAILURE);
+    }
+    pid= waitpid (pid, &status, 0);
+    if(pid==-1){
+        perror("waitpid");    
+    }
 
-/*
- * TODO
- *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
- *   redirect standard out to a file specified by outputfile.
- *   The rest of the behaviour is same as do_exec()
- *
-*/
+    if(WIFEXITED(status)){
+    va_end(args);    
+    if(WEXITSTATUS(status)!=0)
+        return false;
```

Comment:
good... I think you need similar logic in `do_system`.  What happends to `va_end` when you return here though?